test/e2e/kubectl.go

```diff
@@ -112,6 +112,29 @@ var (
 	podProbeParametersVersion = version.MustParse("v1.2.0-alpha.4")
 )
 
+// Stops everything from filePath from namespace ns and checks if everything matching selectors from the given namespace is correctly stopped.
+// Aware of the kubectl example files map.
+func cleanupKubectlInputs(fileContents string, ns string, selectors ...string) {
+	By("using delete to clean up resources")
+	var nsArg string
+	if ns != "" {
+		nsArg = fmt.Sprintf("--namespace=%s", ns)
+	}
+	// support backward compatibility : file paths or raw json - since we are removing file path
+	// dependencies from this test.
+	framework.RunKubectlOrDieInput(fileContents, "delete", "--grace-period=0", "-f", "-", nsArg)
+	for _, selector := range selectors {
+		resources := framework.RunKubectlOrDie("get", "rc,svc", "-l", selector, "--no-headers", nsArg)
+		if resources != "" {
+			framework.Failf("Resources left running after stop:\n%s", resources)
+		}
+		pods := framework.RunKubectlOrDie("get", "pods", "-l", selector, nsArg, "-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ \"\\n\" }}{{ end }}{{ end }}")
```

Comment:
on a heavily loaded cluster even with grace-period of 0 there are killing limits on the kubelet.